### PR TITLE
[NDEV 1061] Fix Navigation Bar Flickering Issue

### PR DESCRIPTION
**Pull request by AdiedX**
## JIRA
https://bypassmobile.atlassian.net/browse/NDEV-1061

## Description
Fix navigation flickering issue when switching between sub tabs

### CODE REVIEW

File: src/js/expanderExpanded.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ function mdExpanderExpandedDirective($mdUtil, $animateCss) {
 
   function link(scope, element, attr, expanderCtrl) {
     var isHeightSet = expanderCtrl.height !== undefined;
+    var isAnimatedOpen = false;
```
Comment (sdrioux) [inline src/js/expanderExpanded.js:26]: 1. Where does this ever get set to true?
2.  Did we not break any tests?

Comment (B-3PO) [reply src/js/expanderExpanded.js:26]: @sdrioux I never wrote tests for this ui component since it was so simple

Comment (AdiedX) [reply src/js/expanderExpanded.js:26]: Updated the PR.

**Comment by B-3PO** [comment]
🚢

File: package.json [hunk 2]
```diff
@@ -14,6 +14,7 @@
     "bump": "^0.2.5",
     "del": "^2.2.0",
     "gulp": "^3.9.1",
+    "gulp-angular-templatecache": "^2.0.0",
```
Comment (JonathanTech) [inline package.json:17]: This was added without anything being added to gulp. Please revert.

File: src/js/expanderExpanded.js (hunk 1 above)
Comment (JonathanTech) [reply src/js/expanderExpanded.js:26]: Let's add tests for this. If it's a simple ui component, then the tests should be easy to write (relatively)

File: src/js/expanderExpanded.js [hunk 3]
```diff
@@ -54,7 +57,7 @@ function mdExpanderExpandedDirective($mdUtil, $animateCss) {
       })
       .start()
       .then(function () {
-        if (isHeightSet !== undefined) {
+        if (isHeightSet === true) {
```
Comment (JonathanTech) [inline src/js/expanderExpanded.js:57]: I really don't see the point of type checking this value. the value will be a bool as defined above

**Review by JonathanTech — CHANGES_REQUESTED** [review]

**Comment by JonathanTech** [comment]
Sorry, there should be tests covering this. Let's get those in :)

File: package.json (hunk 2 above)
Comment (B-3PO) [reply package.json:17]: This change is correct, i forgot to add it when i first built the component
https://github.com/bypasslane/angular-material-expander/blob/95efed78c059bb0bd66a5bdbc8c035ff3cbebcb8/gulpfile.js#L13

File: src/js/expanderExpanded.js (hunk 3 above)
Comment (B-3PO) [reply src/js/expanderExpanded.js:57]: Personally i like to add the true/false so anyone else reading the code can easily see the type, either way is correct.

**Comment by B-3PO** [comment]
@JonathanTech FYI, i am responding to these comments because most of the previous code was written by me

File: package.json (hunk 2 above)
Comment (JonathanTech) [reply package.json:17]: ok, :)

File: src/js/expanderExpanded.js (hunk 3 above)
Comment (JonathanTech) [reply src/js/expanderExpanded.js:57]: To me it communicates that the type might be unknown or there's a possibility of it being not a bool. Personal preferences. Not a show stopper.

Comment (txssseal) [reply src/js/expanderExpanded.js:57]: ya I agree with @JonathanTech.  If we want to do a type check lets wrap it in something like this?

```javascript
if (typeof (variable) === "boolean") {

}
```

**Comment by JonathanTech** [comment]
image
:ship:

**Review by JonathanTech — APPROVED** [review]

**Comment by JonathanTech** [comment]
@B-3PO Could you add me as admin on this project so I can add circle ci?

**Pull request merged**